### PR TITLE
Update to the modern kube-dns; a few other minor tweaks

### PR DESCRIPTION
**Pull request by euank**
Happy to split into two PRs if we want to discuss any of this more, though I think it shouldn't be too bad to just discuss it all here.

cc @joshix @aaronlevy @pbx0

### CODE REVIEW

**Comment by pbx0** [comment]
OK, nice! But two things. First we already have an on-going PR in this vein here: https://github.com/coreos/coreos-kubernetes/pull/603

Its about as far as this PR but, kube-dns 17.1 seems to be the version that matches the v1.3.4 branch we are currently on. So maybe you could coordinate there on who will finish this PR?

Secondly, this is super annoying but we also need the installation tools updated which requires updating:

```
single-node/user-data
multi-node/generic/controller-install.sh
multi-node/generic/worker-install.sh
multi-node/aws/pkg/config/templates/cloud-config-controller
multi-node/aws/pkg/config/templates/cloud-config-worker
```

The first three also actually require you convert the yaml to bash. We normally use:
`python -c "import json,yaml; print json.dumps(yaml.load(open('manifest.yaml')), indent=2, sort_keys=True)"`

**Comment by euank** [comment]
I picked this back up, so I think you owe me some validation @pbx0 :)

I didn't see anything to change in `multi-node/aws/pkg/config/templates/cloud-config-worker` and grepping for `skydns` and `kubedns-amd64:1.3` shows nothing left to change.

**Comment by pbx0** [comment]
Indeed! I have run this through conformance and it looks good. 

LGTM

File: single-node/user-data [hunk 1]
```diff
@@ -448,33 +448,33 @@ EOF
...
-          "version": "v15"
+          "version": "v17.1"
         }
       },
       "spec": {
         "containers": [
           {
             "args": [
-              "--domain=cluster.local.",
+              "--domain=cluster.local",
```
Comment (pbx0) [inline single-node/user-data]: I believe preserving this dot is important.

File: multi-node/aws/pkg/config/templates/cloud-config-controller [hunk 2]
```diff
@@ -455,33 +455,33 @@ write_files:
...
-                  "version": "v15"
+                  "version": "v17.1"
                 }
               },
               "spec": {
                 "containers": [
                   {
                     "args": [
-                      "--domain=cluster.local.",
+                      "--domain=cluster.local",
```
Comment (pbx0) [inline multi-node/aws/pkg/config/templates/cloud-config-controller]: Same here. I think we need the dot.

**Comment by euank** [comment]
Good catch @pbx0, thanks! Updated to include the `.`.

**Comment by pbx0** [comment]
LGTM

**Pull request merged**